lib/kernel-manager.js

```diff
@@ -8,20 +9,35 @@ import path from "path";
 
 import Config from "./config";
 import ZMQKernel from "./zmq-kernel";
+// I imported Kernel so it could be a valid parameter type
+import Kernel from "./kernel.js";
 import KernelPicker from "./kernel-picker";
 import store from "./store";
 import { getEditorDirectory, log } from "./utils";
 
 class KernelManager {
+  // kernelspecs seems to be have this form
+  /*
+  kernelSpec = {
+    display_name: "Existing Kernel",
+    language,
+    argv: [],
+    env: {}
+  };
+  */
+  _kernelSpecs: ?Object;
+  kernelPicker: KernelPicker;
   constructor() {
     this._kernelSpecs = this.getKernelSpecsFromSettings();
   }
 
-  startKernelFor(grammar, onStarted) {
+  startKernelFor(grammar: atom$Grammar, onStarted: (kernel: Kernel) => void) {
     try {
+      // Don't know where atom$Project is and why it doesn't have rootDirectories property
       const rootDirectory = atom.project.rootDirectories[0]
```

Comment:
A better way of doing this would be:
```js
const rootDirectory = atom.project.getPaths()[0] || getEditorDirectory(store.editor);
```
This way we use documented APIs and flow won't complaint 😄